core/src/tasks/helpers.rs

```diff
@@ -240,3 +268,103 @@ fn find_order(base_mint: &BigUint, quote_mint: &BigUint, wallet: &SizedWallet) -
         .find(|(_ind, order)| order.quote_mint.eq(quote_mint) && order.base_mint.eq(base_mint))
         .map(|(ind, _order)| ind)
 }
+
+/// Find a wallet on-chain, and update its validity proofs. That is, a proof of `VALID REBLIND`
+/// for the wallet, and one proof of `VALID COMMITMENTS` for each order in the wallet
+pub(super) async fn update_wallet_validity_proofs(
+    wallet: &Wallet,
+    starknet_client: &StarknetClient,
+    proof_manager_work_queue: CrossbeamSender<ProofManagerJob>,
+    global_state: RelayerState,
+    network_sender: TokioSender<GossipOutbound>,
+) -> Result<(), String> {
+    // No validity proofs needed for an empty wallet, they will be re-proven on
+    // the next update that adds a non-empty order
+    if wallet.orders.values().all(|o| o.is_default()) {
+        return Ok(());
+    }
+
+    // Find the authentication path for the wallet
+    let authentication_path = find_merkle_path(wallet, starknet_client)
+        .await
+        .map_err(|err| err.to_string())?;
+
+    // Dispatch a proof of `VALID REBLIND` for the wallet
+    let (reblind_witness, reblind_response_channel) = construct_wallet_reblind_proof(
+        wallet,
+        authentication_path,
+        proof_manager_work_queue.clone(),
+    )?;
+    let wallet_reblind_witness = Arc::new(reblind_witness);
+
+    // For each order, construct a proof of `VALID COMMITMENTS`
+    let mut commitments_response_channels = Vec::new();
+    for (order_id, order) in wallet.orders.iter().filter(|(_id, o)| !o.is_default()) {
+        // Start a proof of `VALID COMMITMENTS`
+        let (commitments_witness, response_channel) = construct_wallet_commitment_proof(
+            wallet.clone(),
+            order.clone(),
+            proof_manager_work_queue.clone(),
+        )?;
+
+        let order_commitment_witness = Arc::new(commitments_witness);
+
+        // Attach a copy of the witness to the locally managed state
+        // This witness is referenced by match computations which compute linkable commitments
+        // to shared witness elements; i.e. they commit with the same randomness
+        {
+            global_state
+                .read_order_book()
+                .await
+                .attach_validity_proof_witness(
+                    order_id,
+                    OrderValidityWitnessBundle {
+                        reblind_witness: wallet_reblind_witness.clone(),
+                        commitment_witness: order_commitment_witness.clone(),
+                    },
+                )
+                .await;
+        } // order_book lock released
```

Comment:
This is the same as what's used in your `initialize-state` task PR, right? (https://github.com/renegade-fi/renegade/blob/joey/initialize-state/core/src/tasks/initialize_state.rs#L246-L261)

If so, might be nice to pack this into a helper fn

Reply:
I'm not sure I see what a helper would do here? It's a single function call after all